### ISSUE TITLE
support reload for rsyslogd

### ISSUE THREAD

**Issue by Rampa33** [rsyslog/rsyslog#3759]
Today the /etc/init.d/rsyslog reload does not work. Can we have a provision to reload after updating rsyslog.conf file, so that existing TCP based connections are not disturbed.

**Comment by davidelang** [comment]
No, the way that rsyslog handles it's config includes loading new code as part 
of that config (among other problems). It would be a very large task to make it 
so that the config could be recompiled and loaded into a running instance.

David Lang

**Comment by Rampa33** [comment]
Thanks @davidelang for your reply. So if we are doing any changes like adding new remote servers, the idea is to just bear with disconnects and reconnects?

**Comment by davidelang** [comment]
On Mon, 22 Jul 2019, Rampa33 wrote:

> Thanks @davidelang for your reply. So if we are doing any changes like adding 
> new remote servers, the idea is to just bear with disconnects and reconnects?

Yes, if you change anything in the rsyslog.conf you need to do a full shutdown 
and startup. There is talk about being able to do some minor changes without 
having to do a restart, but it's still in very early discussion stages.

David Lang

**Comment by tmccombs** [comment]
Wouldn't logs get lost in this scenario?

**Comment by davidelang** [comment]
On Mon, 10 Aug 2020, Thayne McCombs wrote:

> Wouldn't logs get lost in this scenario?

Yes, unless you are using RELP you will lose some logs when you shutdown.

David Lang

**Comment by AhmedKhaledSaid** [comment]
Does this also apply to logging on the same system ? So for my use case I am logging with my c++ app using syslog on imuxsock,(listening on /dev/log) and writing logs files on the same system.
However I do need to restart rsyslog to reflect config changes, would that cause the logs sent to /dev/log by the c++ app to get lost during the restart ?
